### PR TITLE
Fix: when default visualization is displayed, its entry is not present in the chooser list.

### LINKED ISSUE #5992: Visualization chooser let's me choose Table vis (default one) which causes the vis going blank
### Discord username

_No response_

### What type of issue is this?

Permanent – Ocurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [X] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

https://user-images.githubusercontent.com/12892578/225929654-08fb3d4c-3aaa-4976-a20a-02a5ad2b7c60.mov

### Expected behaviour

I can't choose table visualization when it is a default one

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

126a4668720141e235aaa54c6dad41a10a82ac60

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

The visualization was blank because we set a new visualization instance with same data for attaching, so `VisualizationManager` skipped attaching (and thus the data for previous instance were not shared with the new one).

And the entry was visible, because we informed chooser about selected vis before the visualization list arrived. 

Fixes #5992 may also fix some other issues with blank visualizations.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/view/graph-editor/src/component/visualization/container.rs [hunk 1]
```diff
@@ -526,6 +526,20 @@ impl Container {
         }
 
 
+        // ===  Visualisation Chooser Bindings ===
+
+        frp::extend! { network
+            selected_definition  <- action_bar.visualisation_selection.map(f!([registry](path)
+                path.as_ref().and_then(|path| registry.definition_from_path(path))
+            ));
+            on_selected              <- selected_definition.map(|d|d.as_ref().map(|_|())).unwrap();
```
Comment (vitvakatu) [inline app/gui/view/graph-editor/src/component/visualization/container.rs]:
```suggestion
            on_selected <- selected_definition.map(|d|d.as_ref().map(|_|())).unwrap();
```

**Review by wdanilo — APPROVED** [review]

**Comment by sylwiabr** [comment]
@vitvakatu @MichaelMauderer can one of you QA this?

**Comment by vitvakatu** [comment]
I will check it

**Review by vitvakatu — APPROVED** [review]
QA: passed. Also, I wasn't able to reproduce https://github.com/enso-org/enso/issues/5991 on this branch, but it needs to be checked separately. cc @MichaelMauderer

File: app/gui/view/graph-editor/src/component/visualization/container.rs [hunk 2]
```diff
@@ -526,6 +526,20 @@ impl Container {
         }
 
 
+        // ===  Visualisation Chooser Bindings ===
+
+        frp::extend! { network
+            selected_definition  <- action_bar.visualisation_selection.map(f!([registry](path)
+                path.as_ref().and_then(|path| registry.definition_from_path(path))
+            ));
+            on_selected <- selected_definition.map(|d|d.as_ref().map(|_|())).unwrap();
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/visualization/container.rs]: `.map(|_| ())` -> `.constant(())`

File: app/gui/view/graph-editor/src/component/visualization/container.rs [hunk 3]
```diff
@@ -526,6 +526,20 @@ impl Container {
...
+            eval_ on_selected ( action_bar.hide_icons.emit(()) );
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/visualization/container.rs]: `action_bar.hide_icons <+ on_selected`

File: app/gui/view/graph-editor/src/component/visualization/container.rs [hunk 4]
```diff
@@ -526,6 +526,20 @@ impl Container {
         }
 
 
+        // ===  Visualisation Chooser Bindings ===
+
+        frp::extend! { network
+            selected_definition  <- action_bar.visualisation_selection.map(f!([registry](path)
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/visualization/container.rs]: double space

File: app/gui/view/graph-editor/src/component/visualization/container.rs [hunk 5]
```diff
@@ -538,7 +552,7 @@ impl Container {
         // === Switching Visualizations ===
 
         frp::extend! { network
-            new_vis_definition <- any(frp.set_visualization,vis_after_cycling,default_visualisation);
+            new_vis_definition <- any(frp.set_visualization, selected_definition, vis_after_cycling, default_visualisation).on_change();
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/visualization/container.rs]: > 100 chars

**Review by wdanilo — APPROVED** [review]

**Pull request merged**